### PR TITLE
Feature/rev2 48221 exception handling

### PR DESCRIPTION
**Pull request by saranshvijay**
New Pull Request

### CODE REVIEW

File: src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ConditionWhichThrowsException.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package customexceptions;
+
+public class ConditionWhichThrowsException{
```
Comment (walecharohit) [inline src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ConditionWhichThrowsException.java]: class name seems irrelevant @saranshvijay

File: src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ObservedException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package customexceptions;
+
+public class ObservedException extends Exception{
+	public ObservedException(String info){
```
Comment (walecharohit) [inline src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ObservedException.java]: Very Good !! @saranshvijay

File: src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import customexception.*;
+
+class UncheckedException{
```
Comment (walecharohit) [inline src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java]: change this class name also @saranshvijay this seems to be like more of a exception name

File: src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+class UncheckedException{	
+	public static void main(String [] args){
+	
+	try{
+			ClassWhichThrowsException obj = new ClassWhichThrowsException();
+			obj.conditionChecker(0);
+	}
+			
+	catch(ObservedException e){
```
Comment (walecharohit) [inline src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java]: Do we really need to catch a checked exception ??

File: src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ConditionWhichThrowsException.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+	 
+	  int variable1 = 5;
+	 	public void conditionChecker(int variable2)throws ObservedException{
+			if(variable1 != variable2){
+				throw new ObservedException("The condition does not follow. ");
```
Comment (walecharohit) [inline src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ConditionWhichThrowsException.java]: I can see you are throwing this unchecked exception..where did you throw the checked CheckException's object.

File: src/main/java/com/fnp/checkexception/unchecked-exceptions-program/customexception/ObservedException.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package customexception;
+
+public class ObservedException extends RuntimeException{
```
Comment (walecharohit) [inline src/main/java/com/fnp/checkexception/unchecked-exceptions-program/customexception/ObservedException.java]: Very Good !!

File: src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java (hunk 3 above)
Comment (walecharohit) [reply src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java]: change the name of the folder also which contains this class.

File: src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java (hunk 4 above)
Comment (saranshvijay) [reply src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java]: we have to catch a checked exception but not the unchecked exception.

Comment (walecharohit) [reply src/main/java/com/fnp/checkexception/unchecked-exceptions-program/UncheckedException.java]: yes that's what I am also saying we should not catch it.

File: src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ConditionWhichThrowsException.java (hunk 5 above)
Comment (saranshvijay) [reply src/main/java/com/fnp/checkexception/custom-checked-exception/customexceptions/ConditionWhichThrowsException.java]: In checked exception we have to throw the exception but in the case of unchecked exception it is not necessary to declare in throws
